src/core/src/command/mod.rs

```diff
@@ -29,42 +28,28 @@ pub use self::raw::RawCommandBuffer;
 pub use self::renderpass::*;
 pub use self::transfer::*;
 
-
 /// Thread-safe finished command buffer for submission.
-pub struct Submit<B: Backend, C>(B::SubmitInfo, PhantomData<C>);
+pub struct Submit<B: Backend, C>(pub(crate) B::CommandBuffer, PhantomData<C>);
 unsafe impl<B: Backend, C> Send for Submit<B, C> {}
 
 impl<B: Backend, C> Submit<B, C> {
     ///
-    pub unsafe fn new(info: B::SubmitInfo) -> Self {
-        Submit(info, PhantomData)
-    }
-
-    // Unsafe because we could try to submit a command buffer multiple times.
-    #[doc(hidden)]
-    pub unsafe fn get_info(&self) -> &B::SubmitInfo {
-        &self.0
-    }
-
-    /// Unsafe because we could try to submit a command buffer multiple times by cloning.
-    pub unsafe fn into_info(self) -> B::SubmitInfo {
-        self.0
+    pub unsafe fn new(buffer: CommandBuffer<B, C>) -> Self {
+        Submit(buffer.raw.clone(), PhantomData)
     }
 }
 
 /// Command buffer with compute, graphics and transfer functionality.
-pub struct CommandBuffer<'a, B: 'a + Backend, C> {
-    pub(crate) raw: B::CommandBuffer,
-    pool: &'a mut B::CommandPool,
-    _capability: PhantomData<C>,
+pub struct CommandBuffer<'a, B: Backend, C> {
+    pub(crate) raw: &'a mut B::CommandBuffer,
+    _capability: PhantomData<*const C>,
```

Comment:
why `*const C`?

Reply:
leftover from my attempt to enforce `!Send`, fixed.